imap_processing/glows/tests/test_glows_l1a_data.py

```diff
@@ -0,0 +1,313 @@
+
+    expected_block_header = {
+        "flight_software_version": 131329,
+        "ground_software_version": version,
+        "pkts_file_name": "glows_test_packet_20230920_v00.pkts",
+        "seq_count_in_pkts_file": 0,
+    }
+
+    assert hist.block_header == expected_block_header
+    assert hist.last_spin_id == 0
+
+    assert hist.imap_start_time == TimeTuple(54232215, 0)
+    assert hist.imap_time_offset == TimeTuple(120, 0)
+    assert hist.glows_start_time == TimeTuple(54232214, 1997263)
+    assert hist.glows_time_offset == TimeTuple(119, 1998758)
+    assert hist.flags == {"flags_set_onboard": 64, "is_generated_on_ground": False}
+
+
+def test_direct_event_every_second(decom_test_data):
+    """Test that Level 1A direct events generate correct status data table"""
+    de0 = decom_test_data[1][0]
+
+    # Manually created dictionary from first test packet
+    data_every_second = {
+        "imap_sclk_last_pps": 54232338,
+        "glows_sclk_last_pps": 54232337,
+        "glows_ssclk_last_pps": 1995990,
+        "imap_sclk_next_pps": 54232339,
+        "catbed_heater_active": 0,
+        "spin_period_valid": 1,
+        "spin_phase_at_next_pps_valid": 1,
+        "spin_period_source": 0,
+        "spin_period": 46875,
+        "spin_phase_at_next_pps": 17476,
+        "number_of_completed_spins": 3615489,
+        "filter_temperature": 2507,
+        "hv_voltage": 2007,
+        "glows_time_on_pps_valid": 1,
+        "time_status_valid": 1,
+        "housekeeping_valid": 1,
+        "is_pps_autogenerated": 0,
+        "hv_test_in_progress": 0,
+        "pulse_test_in_progress": 0,
+        "memory_error_detected": 0,
+    }
+
+    de = DirectEventL1A(de0)
+
+    assert dataclasses.asdict(de.status_data) == data_every_second
+
+
+def test_direct_events_uncompressed(decom_test_data):
+    """test building of uncompressed events"""
+    de0 = decom_test_data[1][0]
+    de1a = DirectEventL1A(de0)
+    first_uncompressed_event = bytearray.fromhex("033b8511061e7bf0")
+    processed_uncompressed = de1a._build_uncompressed_event(first_uncompressed_event)
+    expected = DirectEvent(
+        TimeTuple(54232337, 1997808), impulse_length=6, multi_event=False
+    )
+    assert processed_uncompressed == expected
+
+
+def test_direct_events_two_bytes_compressed(decom_test_data):
+    """Test two byte compression for direct events"""
+    de0 = decom_test_data[1][0]
+    de1a = DirectEventL1A(de0)
+    first_byte = int("0x87", 0)
+    two_bytes = bytearray.fromhex("6106")
+
+    expected = DirectEvent(
+        TimeTuple(seconds=54232337, subseconds=1999697),
+        impulse_length=6,
+        multi_event=False,
+    )
+    # Required for processing
+    previous_time = TimeTuple(54232337, 1997808)
+
+    oldest_diff = first_byte & 0x3F
+
+    output = de1a._build_compressed_event(two_bytes, oldest_diff, previous_time)
+
+    final_expected = DirectEvent(
+        TimeTuple(seconds=54232338, subseconds=1988562),
```

Comment:
This was a very good catch! I was indeed not reading the final couple of direct events because I was using the wrong number for the decom steps. I have fixed this issue, and added several additional tests to ensure I am reading all the data as expected, including testing the last direct events in each packet.